git/github.go

```diff
@@ -0,0 +1,85 @@
+package git
+
+import (
+	"bytes"
+	"encoding/json"
+	"fmt"
+	"github.com/weaveworks/flux"
+	"github.com/weaveworks/flux/http/error"
+	"io"
+	"net/http"
+	"net/url"
+)
+
+type Github struct {
+	Client http.Client
+}
+
+type githubResponse struct {
+	Message string `json:"message"`
+}
+
+func (g *Github) InsertDeployKey(inst flux.InstanceID, owner string, repo string, tok string, deployKey string) error {
+	u := url.URL{
+		Scheme: "https",
+		Host:   "api.github.com",
+		Path:   fmt.Sprintf("/repos/%s/%s/keys", owner, repo),
+	}
+
+	// TODO: Check to see if key exists, if so, delete first
+
+	// Encode request, name and key
+	keyReq, err := g.encodeInsertDeployKeyRequest(inst, deployKey)
+
+	// Create key insertion request
+	req, err := http.NewRequest("POST", u.String(), keyReq)
+	if err != nil {
+		return err
+	}
+
+	// Set token in header
+	req.Header.Set("Authorization", fmt.Sprintf("token %s", tok))
+
+	return g.doInsertDeployKeyRequest(req)
+}
+
+func (g *Github) encodeInsertDeployKeyRequest(inst flux.InstanceID, deployKey string) (io.Reader, error) {
+	var keyReq bytes.Buffer
+	keysBody := struct {
+		Title string `json:"title"`
+		Key   string `json:"key"`
+	}{
+		fmt.Sprintf("Flux-%s", inst),
```

Comment:
This is going to end up being something like "Flux-32" (instanceIDs are numbers). It's useful to have an identifier in there for troubleshooting, but I worry about leaking internal identifiers. I don't think we need more than one deploy key in any given repo, do we?